mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerSideEffectTest.java

```diff
@@ -0,0 +1,208 @@
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import javax.mail.Flags;
+
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.MessageManager.FlagsUpdateMode;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.exception.OverQuotaException;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.mailbox.model.Quota;
+import org.apache.james.mailbox.model.QuotaRoot;
+import org.apache.james.mailbox.model.UpdatedFlags;
+import org.apache.james.mailbox.quota.QuotaManager;
+import org.apache.james.mailbox.store.event.MailboxEventDispatcher;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.quota.QuotaImpl;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import com.google.common.collect.ImmutableList;
+
+public abstract class AbstractMessageIdManagerSideEffectTest {
+
+    private static final Quota OVER_QUOTA = QuotaImpl.quota(102, 100);
+    public static final Flags FLAGS = new Flags();
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private MessageIdManager messageIdManager;
+    private MailboxEventDispatcher dispatcher;
+    private MailboxSession session;
+    private Mailbox mailbox1;
+    private Mailbox mailbox2;
+    private QuotaManager quotaManager;
+    private MessageIdManagerTestingData testingData;
+
+    protected abstract MessageIdManagerTestingData createTestingData(QuotaManager quotaManager, MailboxEventDispatcher dispatcher);
+
+    @Before
+    public void setUp() throws Exception {
+        dispatcher = mock(MailboxEventDispatcher.class);
+        session = mock(MailboxSession.class);
+        quotaManager = mock(QuotaManager.class);
+
+        testingData = createTestingData(quotaManager, dispatcher);
+        messageIdManager = testingData.getMessageIdManager();
+        mailbox1 = testingData.getMailbox1();
+        mailbox2 = testingData.getMailbox2();
+
+        when(quotaManager.getMessageQuota(any(QuotaRoot.class))).thenReturn(QuotaImpl.unlimited());
+        when(quotaManager.getStorageQuota(any(QuotaRoot.class))).thenReturn(QuotaImpl.unlimited());
+    }
+
+    @After
+    public void tearDown() {
+        testingData.clean();
+    }
+
+    @Test
+    public void deleteShouldCallEventDispatcher() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+        reset(dispatcher);
```

Comment:
It may have the event when we persist message into "mailbox1" so it needs to reset dispatcher before another action (in this case is delete action) because we just verify event on delete function.